ThunkLibs/libGL/libGL_Guest.cpp

```diff
@@ -8,41 +8,59 @@ desc: Handles glXGetProcAddress
 #define GL_GLEXT_PROTOTYPES 1
 #define GLX_GLXEXT_PROTOTYPES 1
 
-#include "glcorearb.h"
-
 #include <GL/glx.h>
 #include <GL/glxext.h>
 #include <GL/gl.h>
 #include <GL/glext.h>
+
+#undef GL_ARB_viewport_array
+#include "glcorearb.h"
+
 #include <stdio.h>
-#include <cstring>
+#include <cstdlib>
+#include <string_view>
+#include <unordered_map>
 
 #include "common/Guest.h"
 
 #include "thunks.inl"
 #include "function_packs.inl"
 #include "function_packs_public.inl"
+#include "symbol_list.inl"
 
 typedef void voidFunc();
 
+// Maps Vulkan API function names to the address of a guest function which is
+// linked to the corresponding host function pointer
+static std::unordered_map<std::string_view, uintptr_t /* guest function address */> HostPtrInvokers{};
+
 extern "C" {
 	voidFunc *glXGetProcAddress(const GLubyte *procname) {
+    auto Ret = fexfn_pack_glXGetProcAddress(procname);
+    if (!Ret) {
+        return nullptr;
+    }
 
-        for (int i = 0; internal_symtable[i].name; i++) {
-            if (strcmp(internal_symtable[i].name, (const char*)procname) == 0) {
-				// for debugging
-                //printf("glXGetProcAddress: looked up %s %s %p %p\n", procname, internal_symtable[i].name, internal_symtable[i].fn, &glXGetProcAddress);
-                return internal_symtable[i].fn;
-			}
-		}
+    auto TargetFuncIt = HostPtrInvokers.find(reinterpret_cast<const char*>(procname));
```

Comment:
I'd rather us use a flat string list + `std::lower_bound` for binary search here, though that's something we can discuss for a follow up.

Reply:
Yeah, agreed that's an orthogonal refactor better moved to another PR.